lib/buffer.js

```diff
@@ -1277,13 +1280,36 @@ function atob(input) {
       // The first 5 elements of `kForgivingBase64AllowedChars` are
       // ASCII whitespace char codes.
       nonAsciiWhitespaceCharCount++;
+
+      if (index === kEqualSignIndex) {
```

Comment:
If you put this check here, you don't need to repeat the `throw lazyDOMException` part, right?
```suggestion
      if (index === kEqualSignIndex && equalCharCount !== 2) {
```